llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -1429,13 +1430,13 @@ InstructionCost RISCVTTIImpl::getCmpSelInstrCost(unsigned Opcode, Type *ValTy,
       //  vmandn.mm v8, v8, v9
       //  vmand.mm v9, v0, v9
       //  vmor.mm v0, v9, v8
-      return LT.first * 5;
+      return LT.first * TLI->getLMULCost(LT.second) * 5;
```

Comment:
3 of these instructions are mask instructions that only operate on at most VLEN bits. They shouldn't scale with LMUL.